ci-operator/step-registry/hypershift/mce/agent/create/agentserviceconfig/baremetal/hypershift-mce-agent-create-agentserviceconfig-baremetal-commands.sh

```diff
@@ -6,7 +6,7 @@ set -o pipefail
 set -E
 
 export PATH=${PATH}:/cli
-gnu_architecture=$(sed 's/amd64/x86_64/;s/arm64/aarch64/' <<< "${architecture:-amd64}")
+gnu_architecture=${AGENTSERVICECONFIG_OSIMAGE_ARCH}
```

Comment:
gnu_architecture=$(sed 's/amd64/x86_64/;s/arm64/aarch64/' <<< "${architecture:-amd64}")
hypershift_architecture_workaround=$(sed 's/amd64/x86_64/;s/aarch64/arm64/' <<< "${architecture:-amd64}")

Reply:
use `hypershift_architecture_workaround` in lines 80, 81

Reply:
maybe we'd raise a bug about this. The architectures in the file https://github.com/openshift/assisted-service/blob/master/data/default_os_images.json#L100 are mixed GNU (for x86) and LLVM (for arm).

cc @jeffdyoung @bryan-cox